src/NUnitFramework/tests/Attributes/TimeoutTests.cs

```diff
@@ -120,19 +121,41 @@ public void TestTimeOutElapsed()
             Assert.That(result.Message, Does.Contain("100ms"));
         }
 
-        // TODO: The test in TimeoutTestCaseFixture work as expected when run
-        // directly by NUnit. It's only when run via TestBuilder as a second
-        // level test that the result is incorrect. We need to fix this.
-        [Test, Explicit("Timing issue")]
-        public void TestTimeOutTestCaseWithOutElapsed()
+        [Explicit("Tests that demonstrate Timeout failure")]
+        public class ExplicitTests
         {
-            TimeoutTestCaseFixture fixture = new TimeoutTestCaseFixture();
-            TestSuite suite = TestBuilder.MakeFixture(fixture);
-            ParameterizedMethodSuite testMethod = (ParameterizedMethodSuite)TestFinder.Find("TestTimeOutTestCase", suite, false);
-            ITestResult result = TestBuilder.RunTest(testMethod, fixture);
-            Assert.That(result.ResultState, Is.EqualTo(ResultState.Failure), "Suite result");
-            Assert.That(result.Children.ToArray()[0].ResultState, Is.EqualTo(ResultState.Success), "First test");
-            Assert.That(result.Children.ToArray()[1].ResultState, Is.EqualTo(ResultState.Failure), "Second test");
+            [Test, Timeout(50)]
+            public void TestTimesOut()
+            {
+                while (true) ;
+            }
+
+            [Test, Timeout(50), RequiresThread]
+            public void TestTimesOutUsingRequiresThread()
+            {
+                while (true) ;
+            }
+
+            [Test, Timeout(50), Apartment(ApartmentState.STA)]
+            public void TestTimesOutInSTA()
+            {
+                while (true) ;
```

Comment:
Suggestion: could these instead throw `Assert.Fail` after 100ms or something? Slightly concerned that the result of this breaking is the test suite hangs - which would prevent a merge, but wouldn't be so easy to debug!